File: src/main/java/com/antwerkz/ophelia/dao/UserFinder.java
package com.antwerkz.ophelia.dao;

import com.antwerkz.ophelia.models.User;
import org.mongodb.morphia.Datastore;

public class UserFinder extends OpheliaDao<User> {
  public UserFinder(Datastore ds) {
    super(ds, User.class);
  }

  public boolean initialized() {
    return getDs().createQuery(User.class).get() != null;
  }
}

File: src/main/kotlin/com/antwerkz/ophelia/dao/UserFinder.kt
package com.antwerkz.ophelia.dao

import com.antwerkz.ophelia.models.User
import org.mongodb.morphia.Datastore

public class UserFinder(ds: Datastore) : OpheliaDao<User>(ds, javaClass<User>()) {

    public fun initialized(): Boolean {
        return ds.createQuery<User>(javaClass<User>()).get() != null
    }
}
